tools/emscripten.py

```diff
@@ -507,19 +513,20 @@ def finalize_wasm(infile, outfile, js_syms):
       args.append('--dyncalls-i64')
       # we need to add some dyncalls to the wasm
       modify_wasm = True
+
   if settings.AUTODEBUG:
     # In AUTODEBUG mode we want to delay all legalization until later.  This is hack
     # to force wasm-emscripten-finalize not to do any legalization at all.
     args.append('--bigint')
-  else:
-    if settings.LEGALIZE_JS_FFI:
-      # When we dynamically link our JS loader adds functions from wasm modules to
-      # the table. It must add the original versions of them, not legalized ones,
-      # so that indirect calls have the right type, so export those.
-      args += building.js_legalization_pass_flags()
-      modify_wasm = True
-    else:
-      args.append('--no-legalize-javascript-ffi')
+    legalizing = False
+
+  if not settings.LEGALIZE_JS_FFI:
```

Comment:
Can we not use `settings.LEGALIZE_JS_FFI` where the new `legalizing` variable is set? That is, I'm not sure why a new variable is needed here.

Reply:
The reason is that `LEGALIZE_JS_FFI` does not mean "don't do any legalization at all", it just means "do partial legalization".

See https://github.com/WebAssembly/binaryen/blob/2471301a5209724b1ea32fab36b13410e96c0af9/src/tools/wasm-emscripten-finalize.cpp#L262-L267

At least today.

We could change that but I think we can/should consider that separately.